src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,146 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        Node(T value) {
+            this.value = value;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (size == 0) {
+            first = newNode;
+            last = newNode;
+        } else {
+            last.next = newNode;
+            newNode.prev = last;
+            last = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Invalid index");
```

Comment:
```suggestion
            throw new IndexOutOfBoundsException("Invalid index: " + index);
```
make message more informative